asmap-tool.py

```diff
@@ -145,7 +145,14 @@ def main():
                 print("%s AS%i # was unassigned" % (net, new_asn))
             else:
                 print("%s AS%i # was AS%i" % (net, new_asn, old_asn))
-        print("# %i (2^%f) IPv4 addresses changed; %i (2^%f) IPv6 addresses changed" % (ipv4_changed, math.log2(ipv4_changed), ipv6_changed, math.log2(ipv6_changed)))
+        if ipv4_changed > 0 and ipv6_changed == 0:
+            print("# %i (2^%.2f) IPv4 addresses changed; %i IPv6 addresses changed" % (ipv4_changed, math.log2(ipv4_changed), ipv6_changed))
+        elif ipv6_changed > 0 and ipv4_changed == 0:
+            print("# %i IPv4 addresses changed; %i (2^%.2f) IPv6 addresses changed" % (ipv4_changed, ipv6_changed, math.log2(ipv6_changed)))
+        elif ipv6_changed == 0 and ipv4_changed == 0:
+            print("# %i IPv4 addresses changed; %i IPv6 addresses changed" % (ipv4_changed, ipv6_changed))
+        else:
+            print("# %i IPv4 addresses changed; %i IPv6 addresses changed" % (ipv4_changed, math.log2(ipv4_changed), ipv6_changed, math.log2(ipv6_changed)))
```

Comment:
I think we could look at using conditional expressions here to avoid repetition.
(untested):
```suggestion
        print(
            "# %i%s IPv4 addresses changed; %i%s IPv6 addresses changed"
            % (
                ipv4_changed,
                "" if ipv4_changed == 0 else " (2^%.2f)" % math.log2(ipv4_changed),
                ipv6_changed,
                "" if ipv6_changed == 0 else " (2^%.2f)" % math.log2(ipv6_changed),
            )
        )

```

Reply:
Cool, cleaner. Tested it and worked.